src/components/config/index.ts

```diff
@@ -138,3 +150,25 @@ export const setupConfigFromUrl = async (
     updateConfig(fetched)
   }, checkingInterval * 1000)
 }
+
+export const setupConfig = async (flags: any): Promise<Config | undefined> => {
+  if (flags['create-config'] && !flags.har && !flags.postman) {
+    log.info(
+      'Opening Monika Configuration Generator in your default browser...'
+    )
+    open('https://hyperjumptech.github.io/monika-config-generator/')
+    return
+  }
```

Comment:
One thing that is for certain: **When `create-config` exists, monika does not run. It either opens the browser or generate a config file from postman or har.** So, it's simpler to put the logic here:

```javascript
// extract this as a function so can be called by other functions
const getPathAndTypeFromFlags = (flags: any): { path: string; type: string } => {
  let path = flags.config
  let type = 'monika'

  if (flags.postman) {
    path = flags.postman
    type = 'postman'
  }
  if (flags.har) {
    path = flags.har
    type = 'har'
  }

  return {
    path,
    type,
  }
}

export const createConfig = async (flags: any) => {
  if (!flags.har && !flags.postman) {
    log.info(
      'Opening Monika Configuration Generator in your default browser...'
    )
    open('https://hyperjumptech.github.io/monika-config-generator/')
  } else {
    const { path, type } = getPathAndTypeFromFlag(flags)

    const parsed = parseConfig(path, type)
    writeConfigFile(flags, parsed)
  }
}

export const setupConfig = async (flags: any) => {
  if (isUrl(flags.config)) {
    await setupConfigFromUrl(flags.config, flags['config-interval'])
  } else {
    const watchConfigFile = !(
      process.env.CI ||
      process.env.NODE_ENV === 'test' ||
      flags.repeat
    )

    await setupConfigFromFile(flags, watchConfigFile)
  }
}
```

With this, 
1. we can call `createConfig` from `src/index.ts` as early as possible, because this function does not require any db setup or anything else.
2. `setupConfig` function is responsible only for running monika using existing configuration file.

Reply:
fixed refactor